src/quantum/azext_quantum/operations/job.py

```diff
@@ -0,0 +1,240 @@
+    # Can't call run directly because it fails to understand the
+    # `ExecutionTarget` property when passed in the command line
+    if not no_build:
+        build(cmd, target_id=target_id, project=project)
+
+    logger.info("Project built successfully.")
+
+    ws = WorkspaceInfo(cmd, resource_group_name, workspace_name, location)
+    target = TargetInfo(cmd, target_id)
+    token = _get_data_credentials(cmd.cli_ctx, ws.subscription).get_token().token
+
+    args = _generate_submit_args(program_args, ws, target, token, project, job_name, shots, storage)
+
+    import subprocess
+    result = subprocess.run(args, stdout=subprocess.PIPE, check=False)
+
+    if result.returncode == 0:
+        job_id = result.stdout.decode('ascii').strip()
+        return show(cmd, job_id, resource_group_name, workspace_name)
+
+    raise CLIError("Failed to submit job.")
+
+
+def _parse_blob_url(url):
+    from urllib.parse import urlparse
+    o = urlparse(url)
+
+    account_name = o.netloc.split('.')[0]
+    container = o.path.split('/')[-2]
+    blob = o.path.split('/')[-1]
+    sas_token = o.query
+
+    return {
+        "account_name": account_name,
+        "container": container,
+        "blob": blob,
+        "sas_token": sas_token
+    }
+
+
+def output(cmd, job_id, resource_group_name=None, workspace_name=None, location=None):
+    """
+    Get the results of running a Q# job.
+    """
+    import tempfile
+    import json
+    import os
+    from azure.cli.command_modules.storage._client_factory import blob_data_service_factory
+
+    path = os.path.join(tempfile.gettempdir(), job_id)
+
+    if os.path.exists(path):
+        logger.debug("Using existing blob from %s", path)
+    else:
+        logger.debug("Downloading job results blob into %s", path)
+
+        info = WorkspaceInfo(cmd, resource_group_name, workspace_name, location)
+        client = cf_jobs(cmd.cli_ctx, info.subscription, info.resource_group, info.name, info.location)
+        job = client.get(job_id)
+
+        if job.status != "Succeeded":
+            return f"Job status: {job.status}. Output only available if Succeeded."
+
+        args = _parse_blob_url(job.output_data_uri)
+        blob_service = blob_data_service_factory(cmd.cli_ctx, args)
+        blob_service.get_blob_to_path(args['container'], args['blob'], path)
+
+    with open(path) as json_file:
+        data = json.load(json_file)
+        return data
+
+
+def wait(cmd, job_id, resource_group_name=None, workspace_name=None, location=None, max_poll_wait_secs=5):
+    """
+    Place the CLI in a waiting state until the job finishes running.
+    """
+    import time
+
+    def has_completed(job):
+        return job.status in ("Succeeded", "Failed", "Cancelled")
+
+    info = WorkspaceInfo(cmd, resource_group_name, workspace_name, location)
+    client = cf_jobs(cmd.cli_ctx, info.subscription, info.resource_group, info.name, info.location)
+
+    # TODO: LROPoller...
+    w = False
```

Comment:
What is `w`? Magic name.

Reply:
Ha, this code has been here for a while. I know what Andres intended. I'll make the code more clear here, with better name and comments, but I'll address that TODO in a later rev.

Reply:
Thanks. Tiny problem. It doesn't matter.